chain/client/src/sync.rs

```diff
@@ -166,12 +195,19 @@ impl HeaderSync {
             true
         } else {
             // Resetting the timeout as long as we make progress.
-            if header_head.height > latest_height {
-                self.prev_header_sync = (
-                    now + Duration::seconds(BLOCK_HEADER_PROGRESS_TIMEOUT),
-                    header_head.height,
-                    prev_height,
+            let ns_time_till_timeout =
+                (to_timestamp(timeout).saturating_sub(to_timestamp(now))) as u128;
+            let remaining_expected_weight =
+                self.expected_weight_per_second * ns_time_till_timeout / 1_000_000_000u128;
```

Comment:
Are we sure that this is not going to overflow?

Reply:
Yes, will add a comment